### ISSUE TITLE
Fabric 2.4 binding can get stuck waiting for unfinished transactions

### ISSUE THREAD

**Issue by davidkel** [hyperledger/caliper#1340]
On running tests on occasion the following can happen
```
2022.05.13-13:06:13.776 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:18.777 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:23.778 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:28.779 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:33.780 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:38.781 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:43.782 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:48.784 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:53.786 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:06:58.787 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:07:03.789 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
2022.05.13-13:07:08.790 info  [caliper] [default-observer]      [get-asset-evaluate-100 Round 3 Transaction Info] - Submitted: 21 Succ: 20 Fail:0 Unfinished:1
```

This also happens for transactions that are submitted. As this example is evaluate it looks like the evaluate call is never unblocked even though a timeout of 60 seconds is given to evaluate it appears to not fire and so caliper hangs waiting for an unfinished transaction which never completes

This isn't likely to be an issue with caliper or the connector implementation however it causes caliper a lot of problems if it does occur (and it's also been seen in other scenarios).

There are several options here
1. re-think the need for a caliper-worker to wait for unfinished transactions or put a time limit on it
2. have the base connector implement it's own timeout on submit calls
3. have the fabric connectors set a timer around the call as well as use the timers from the sdk so the connector can ensure it really does timeout

I was able to recreate this quite frequently on grpc-js 1.6.7 and seems to be less frequently on 1.4.2. I was also running caliper in the current docker container setup of `node:14.19-alpine`  if I added a console.log to caliper-worker when it submits a transaction in the SetImmediatePromise call then I couldn't recreate the problem. It's very odd and going to be a nightmare to track down is it is so intermittent

**Comment by davidkel** [comment]
Ok, I think I may have found the issue which would affect all SUTs. txFinished in the transaction-statistics-collector has a situation where it won't mark the transaction as finished, also the time comparison should be >=, not > as time granularity in node is not going to be good enough on a fast machine.

**Comment by davidkel** [comment]
Also found a big potential issue with the implementation of the TxObserverInterface, so will raise an issue for that but doesn't have to be fixed for 0.5.0

**Comment by Pra-San** [comment]
Hi I am still facing the same issue 2 years later on the latest caliper version and latest fabric version.

`2024.05.04-08:27:49.706 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:27:54.706 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:27:59.708 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:04.708 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:09.709 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:14.710 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:19.712 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:24.713 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:29.714 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:34.715 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:39.717 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
2024.05.04-08:28:44.717 info  [caliper] [default-observer] 	[Read asset fixed-rate-2 Round 3 Transaction Info] - Submitted: 601 Succ: 600 Fail:0 Unfinished:1
`
